suse_migration_services/units/reboot.py

```diff
@@ -25,6 +27,22 @@ def main():
     DistMigration reboot with new kernel
     """
     try:
+        # Note:
+        # After the migration process is finished, the system reboots.
+        # The system will boot either in a migrated system or in a
+        # roll-backed system one in case the migration failed.
+        # In order to avoid rebooting, the migration must have failed
+        # and /etc/system_migration_debug file must exist.
+        debug_file = '/etc/system_migration_debug'
+        if os.path.exists(debug_file):
+            log.info('There is a debug file')
+            migration_has_failed = check_failed_migration()
+            if migration_has_failed:
+                log.info('Migration failed. Not rebooting')
```

Comment:
I question the logging here. Your log entries are two fold:

```
log.info('There is a debug file')
log.info('Migration failed. Not rebooting')
```

If the migration hasn't failed we will get a log file entry that tells us that we have a debug file, nice but pointless imho. Why not having one log message in case the migration failed and we instructed the system to not reboot

```
log.warning('Migration process failed, reboot skipped due to debug flag set')
```

or similar

Reply:
Yes, that sounds better